lib/Doctrine/ORM/Query/TreeWalkerChain.php

```diff
@@ -33,6 +33,7 @@ class TreeWalkerChain implements TreeWalker
      * The tree walkers.
      *
      * @var TreeWalker[]
+     * @psalm-var TreeWalkerChainIterator<TreeWalker>
```

Comment:
This looks repetitive. I think it would be better to use just `TreeWalkerChainIterator`, and use `class-implements Iterator<TreeWalker>` in `TreeWalkerChainIterator`

Reply:
You're right. Actually, I have pretty much this in a changeset. I wasn't sure it was a good idea to bring up template for generics so soon :)